pilot/ARCHITECTURE.md

```diff
@@ -0,0 +1,181 @@
+# Architecture of Istiod
+
+This document describes the high level architecture of the Istio control plane, Istiod.
+Istiod is structured as a modular monolith, housing a wide range of functionality from certificate signing, proxy configuration (XDS), traditional Kubernetes controllers, and more.
+
+## Proxy Configuration
+
+Istiod's primary role - and most code - is to dynamically configure proxies (Envoy sidecars and ingress, gRPC, ztunnel, and more). This roughly consists of 3 parts:
+1. Config ingestion (inputs to the system)
+1. Config translation
+1. Config serving (XDS)
+
+### Config Ingestion
+
+Istio reads from over 20 different resources types, and aggregates them together to build the proxy configuration. These resources can be sourced from Kubernetes (via watches), files, or over xDS; Kubernetes is by far the most common usage, though.
+
+Primarily for historical reasons, ingestion is split into a few components.
```

Comment:
Not sure if only historical - but I think the design was that we generally distinguish 'configs' and 'load balancing'/routing, with the later ( endpoint watching and EDS) having different path. Maybe also mention LB and routing have different requirements - fast and frequent but smaller in size- while config updates can be large and may be throttled/delayed more.